src/renderers/dom/client/wrappers/ReactDOMInput.js

```diff
@@ -48,6 +48,11 @@ function warnIfValueIsNull(props) {
   }
 }
 
+function isControlled(props) {
+  return ((props.type === 'checkbox' || props.type === 'radio') && props.checked !== undefined) ||
+    (props.type !== 'checkbox' && props.type !== 'radio' && props.value !== undefined);
+}
```

Comment:
This spacing isn't FB style and this would be clearer anyway; can you change it?

```
function isControlled(props) {
  var usesChecked = props.type === 'checkbox' || props.type === 'radio';
  return usesChecked ? props.checked !== undefined : props.value !== undefined;
}
```